lib/domain/usecase/search/server_search_interactor.dart

```diff
@@ -11,16 +11,18 @@ class ServerSearchInteractor {
       getIt.get<ServerSearchRepository>();
 
   Stream<Either<Failure, Success>> execute({
-    String? nextBatch,
+    ServerSearchChatSuccess? lastSuccess,
     required String searchTerm,
     List<KeyKind>? keys = const [KeyKind.contentBody],
     SearchOrder orderBy = SearchOrder.recent,
     List<GroupKey> groupKeys = const [GroupKey.roomId],
   }) async* {
     try {
-      yield Right(ServerSearchInitial());
+      if (lastSuccess == null) {
```

Comment:
Why not add SeẻverSearchLoading state here

Reply:
you are author of this line before

Reply:
I know, i means when lastSearch is not null, yield loading state

Reply:
you still need keep current state when loading more, so dont yield new state in this case